test/integration/upgen_int_test.go

```diff
@@ -52,8 +57,8 @@ func (suite *UpdateGenIntegrationTestSuite) TestUpdateBits() {
 
 	cp.ExpectExitCode(0)
 
-	stateExec, err := installation.StateExecFromDir(filepath.Join(tempPath, constants.ToplevelInstallArchiveDir))
-	suite.NoError(err)
+	baseDir := filepath.Join(tempPath, constants.ToplevelInstallArchiveDir)
+	stateExec := filepath.Join(baseDir, installation.BinDirName, constants.StateCmd+osutils.ExeExt)
```

Comment:
This should work with `installation.StateExecFromDir` and I'm pretty sure it was in the PR that moved appinfo. What changed?

Reply:
That function is for using information from the installed state tool. This test is using a generated payload. While those are similar things they are not the same.

I believe in this case it would fail because the payload has no install dir marker.